crypto/rand/rand_lib.c

```diff
@@ -465,7 +493,8 @@ void rand_pool_free(RAND_POOL *pool)
     if (pool == NULL)
         return;
 
-    OPENSSL_secure_clear_free(pool->buffer, pool->max_len);
+    if (!pool->attached)
+        OPENSSL_secure_clear_free(pool->buffer, pool->max_len);
```

Comment:
Does it make sense here to clear an attached pool?
Or should that be left to the user?

Reply:
From a security viewpoint it would make sense, but it would break the API, because the buffer is declared const for `RAND_bytes()` and `RAND_add()`. Although it was necessary to cast the `const` away internally, its content needs to remain untouched.

Reply:
One reason more why the entropy callback mechanism is superiour to `RAND_add()` and `RAND_seed()`...

Reply:
> Does it make sense here to clear an attached pool?

I'll better add a comment that this should not be done...